plugins/auto-sizes/hooks.php

```diff
@@ -100,16 +119,64 @@ function auto_sizes_render_generator(): void {
  * @param array<string, mixed> $parsed_block The parsed block.
  * @return string The updated block content.
  */
-function auto_sizes_improve_image_sizes_attribute( string $content, array $parsed_block ): string {
-
+function auto_sizes_filter_image_tag( string $content, array $parsed_block ): string {
 	$processor = new WP_HTML_Tag_Processor( $content );
 	$has_image = $processor->next_tag( array( 'tag_name' => 'img' ) );
 
 	// Only update the markup if an image is found.
 	if ( $has_image ) {
+		$align = $parsed_block['attrs']['align'] ?? 'default';
+
+		if ( $align ) {
+			$processor->set_attribute( 'data-align', $align );
+		}
+
+		// Resize image width.
+		$resize_image_width = $parsed_block['attrs']['width'] ?? '';
+		if ( $resize_image_width ) {
+			$processor->set_attribute( 'data-resize-width', $resize_image_width );
+		}
+
+		$content = $processor->get_updated_html();
+	}
+	return $content;
+}
+add_filter( 'render_block_core/image', 'auto_sizes_filter_image_tag', 10, 2 );
+add_filter( 'render_block_core/cover', 'auto_sizes_filter_image_tag', 10, 2 );
+
+/**
+ * Filter the sizes attribute for images to improve the default calculation.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $content The block content about to be rendered.
+ * @return string The updated block content.
+ */
+function auto_sizes_improve_image_sizes_attributes( string $content ): string {
+	$processor = new WP_HTML_Tag_Processor( $content );
+	while ( $processor->next_tag( array( 'tag_name' => 'img' ) ) ) {
+
+		$has_auto_sizes = false;
+		if ( str_contains( $content, 'auto,' ) ) {
+			$has_auto_sizes = true;
+		}
```

Comment:
Good idea, but we should be more specific here and only check the `sizes` attribute. I wonder if we avoid the need for this check by making sure this filter runs prior to the `auto_sizes_update_content_img_tag()` callback?

```suggestion
		$old_sizes      = $processor->get_attribute( 'sizes' );
		$has_auto_sizes = str_starts_with( $old_sizes, 'auto' );
```

Reply:
Make sense. I have added higher priority for sizes filter and remove these code as now it not needed